dataactcore/utils/jobQueue.py

```diff
@@ -1,120 +1,158 @@
+from contextlib import contextmanager
+from csv import reader
+
 from celery import Celery
-from dataactcore.config import CONFIG_DB, CONFIG_SERVICES, CONFIG_JOB_QUEUE, CONFIG_BROKER
+
+from dataactcore.config import (
+    CONFIG_DB, CONFIG_SERVICES, CONFIG_JOB_QUEUE, CONFIG_BROKER)
 import requests
 from dataactvalidator.filestreaming.csvS3Writer import CsvS3Writer
 from dataactvalidator.filestreaming.csvLocalWriter import CsvLocalWriter
-from csv import reader
 from dataactcore.utils.jsonResponse import JsonResponse
+from dataactcore.utils import fileF
 from dataactcore.utils.responseException import ResponseException
 from dataactcore.utils.statusCode import StatusCode
 
+
+def enqueue(jobID):
+    """POST a job to the validator"""
+    validatorUrl = '{validator_host}:{validator_port}'.format(
+        **CONFIG_SERVICES)
+    if 'http://' not in validatorUrl:
+        validatorUrl = 'http://' + validatorUrl
+    validatorUrl += '/validate/'
+    params = {
+        'job_id': jobID
+    }
+    response = requests.post(validatorUrl, params)
+    return response.json()
+
+
+def get_xml_response_content(api_url):
+    """ Retrieve XML Response from the provided API url """
+    return requests.get(api_url, verify=False).text
+
+
+def download_file(local_file_path, file_url):
+    """ Download a file locally from the specified URL """
+    with open(local_file_path, "w") as file:
+        # get request
+        response = requests.get(file_url)
+        # write to file
+        response.encoding = "utf-8"
+        file.write(response.text)
+
+
+def get_lines_from_csv(file_path):
+    """ Retrieve all lines from specified CSV file """
+    lines = []
+    with open(file_path) as file:
+        for line in reader(file):
+            lines.append(line)
+    return lines
+
+
+def write_csv(user_id, file_name, is_local, header, body):
+    """Derive the relevant location and write a CSV to it.
+    :return: the final file name (complete with prefix)"""
+    if is_local:
+        file_name = CONFIG_BROKER['broker_files'] + file_name
+        csv_writer = CsvLocalWriter(file_name, header)
+    else:
+        file_name = "{}/{}".format(user_id, file_name)
+        bucket = CONFIG_BROKER['aws_bucket']
+        region = CONFIG_BROKER['aws_region']
+        csv_writer = CsvS3Writer(region, bucket, file_name, header)
+
+    with csv_writer as writer:
+        for line in body:
+            writer.write(line)
+        writer.finishBatch()
+    return file_name
+
+
+@contextmanager
```

Comment:
The `contextmanager` decorator is a shorthand for creating "contexts" (`with` statements). To create something like
```python
with mycontext() as ctx:
    do_something_with(ctx)
```
we can either create an object with `__enter__` and `__exit__` magic methods or we can use the `contextmanager` decorator. These two examples are more or less equivalent:

```python
class mycontext:
    def __enter__(self):
        some_setup_here()
        return "context"

    def __exit__(self):
        some_teardown_here()


@contextmanager
def mycontext():
    some_setup_here()
    yield "context"
    some_teardown_here()
```
As you can see, the later's much easier to read.